packages/react-dom/src/client/ReactDOMHostConfig.js

```diff
@@ -247,9 +246,21 @@ export function getPublicInstance(instance: Instance): * {
   return instance;
 }
 
-export function prepareForCommit(containerInfo: Container): void {
+export function prepareForCommit(
+  containerInfo: Container,
+  isInstanceInsideHiddenOrRemovedTree: (instance: Object) => boolean,
+): void {
   eventsEnabled = ReactBrowserEventEmitterIsEnabled();
   selectionInformation = getSelectionInformation();
+  if (enableDeprecatedFlareAPI || enableUseEventAPI) {
+    const focusedElem = selectionInformation.focusedElem;
+    if (focusedElem !== null) {
+      const instance = getClosestInstanceFromNode(focusedElem);
+      if (instance !== null && isInstanceInsideHiddenOrRemovedTree(instance)) {
+        dispatchBeforeDetachedBlur(focusedElem);
+      }
+    }
+  }
```

Comment:
Do we need to add a `try`/`finally` now (in case the code goes down the path of `DEPRECATED_dispatchEventForResponderEventSystem`)

Reply:
Thankfully we don't need it. User code is wrapped with a `try/catch/finally` block that we use from the legacy event system.